sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/ExceptionMappingPolicy.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.models;
+
+import com.azure.core.exception.HttpResponseException;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.storage.common.Utility;
+import reactor.core.publisher.Mono;
+
+/**
+ * This is a exception mapping policy offers functionality to mapping the error to specific exception.
+ */
+public class ExceptionMappingPolicy implements HttpPipelinePolicy {
+
+    /**
+     * Applies cleaning to eTag returned from the service if the request was successful, otherwise maps the returned
+     * {@link StorageErrorException} to a {@link StorageException} which promotes information from the returned error.
+     *
+     * @return The cleansed success response or the mapped exception
+     */
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        return next.clone().process()
+            .onErrorResume(StorageErrorException.class, resume ->
+                    resume.getResponse()
+                        .getBodyAsString()
+                        .switchIfEmpty(Mono.just(""))
+                        .flatMap(body -> Mono.error(new StorageException(resume, body)))
```

Comment:
same comment here as above w.r.t empty vs null body.

Reply:
Deleted the policy.  For the ScrubEtagPolicy, put it in common policy.